src/printer.js

```diff
@@ -3297,6 +3303,19 @@ function printArrayItems(path, options, printPath, print) {
   return concat(printedElements);
 }
 
+function isParserKeepingEntities(node, options) {
```

Comment:
I'm okay with this approach, but here's how I would like it. Have a function `getJSXRawValue(node)` that checks if `node.extra.rawValue` or `node.raw` exists and return it, otherwise read from the original source.